### PR TITLE
refactor: Add structure classes to qubit4 parser

### PR DESCRIPTION
**Pull request by nathan-stender**

### CODE REVIEW

File: src/allotropy/parsers/utils/values.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ def quantity_or_none(
         return None
     if isinstance(value, list):
         return value_cls(value=value[assert_not_none(index, msg="Cannot provide list to quantity_or_none without index")])  # type: ignore[call-arg]
+    # Typing does not know that all subclasses of TQuantityValue have default value for unit set.r
```
Comment (mdlu) [inline src/allotropy/parsers/utils/values.py:108]: typo here I think, what should the end say?

File: src/allotropy/parsers/thermo_fisher_qubit4/thermo_fisher_qubit4_parser.py [hunk 2]
```diff
@@ -216,170 +94,94 @@ def _get_model(self, data: pd.DataFrame, filename: str) -> Model:
         )
 
     def _get_spectrophotometry_document(
-        self, data: pd.DataFrame
-    ) -> list[SpectrophotometryDocumentItem]:
+        self, row: Row
+    ) -> SpectrophotometryDocumentItem:
         """
         Generates a list of spectrophotometry document items from the given data.
```
Comment (mdlu) [inline src/allotropy/parsers/thermo_fisher_qubit4/thermo_fisher_qubit4_parser.py:100]: should update this if it's not a list anymore

File: src/allotropy/parsers/thermo_fisher_qubit4/thermo_fisher_qubit4_structure.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+class Row:
+    timestamp: str
+    assay_name: str | None
+    fluorescence: float
+    batch_identifier: str | None
+    sample_identifier: str
+    sample_volume: float | None
+    excitation: str | None
+    emission: str | None
+    diluation_factor: float | None
```
Comment (mdlu) [inline src/allotropy/parsers/thermo_fisher_qubit4/thermo_fisher_qubit4_structure.py:32]: small typo: `dilution_factor`

File: src/allotropy/parsers/utils/values.py (hunk 1 above)
Comment (nathan-stender) [reply src/allotropy/parsers/utils/values.py:108]: Oh that end `r` is the typo. Fixed.

**Review by mdlu — APPROVED** [review]

**Pull request merged**